File: AdvancePython/Day-3 Advanced OOP concepts/Task-3-3.py
# Task 3:
# 3. Method overloading:
def add_values(*args: any) -> None:
    if not args:
        print("No argument provided!")
        return

    first_arg = args[0]

    if isinstance(first_arg, int):
        result = 0
    elif isinstance(first_arg, str):
        result = ""
    else:
        print(f"Unsupported datatype: {type(first_arg).__name__}")
        return

    try:
        for x in args:
            result += x
        print(result)
    except TypeError:
        print(f"Error: all arguments must be of the same type.")


if __name__ == "__main__":
    add_values(5, 6)
    add_values("Hello ", "World")
    add_values(1.2, 1.4)
    add_values(10, "mixed")

File: node.js/Day-3/Task-3-3.js
// Task 3:
// 3. Method overloading:

function Add(datatype, ...args) {
    let answer;

    if(datatype == 'int') {
        answer = 0;
    }

    if(datatype == 'str') {
        answer = "";
    }

    for (let x = 0; x < args.length; x++) {
        answer += args[x];
    }
    console.log(answer);
}

Add('int', 5, 5)
Add('str', "Hello ", "World");